data_processing.ipynb

```diff
@@ -748,6 +820,726 @@
+      "Epoch 18/20\n",
+      "3/3 [==============================] - 13s 4s/step - loss: 24.9505 - accuracy: 0.5861 - val_loss: 32.8662 - val_accuracy: 0.3706\n",
+      "Epoch 19/20\n",
+      "3/3 [==============================] - 14s 5s/step - loss: 14.4200 - accuracy: 0.6103 - val_loss: 42.7369 - val_accuracy: 0.3357\n",
+      "Epoch 20/20\n",
+      "3/3 [==============================] - 10s 3s/step - loss: 13.5930 - accuracy: 0.6224 - val_loss: 25.0608 - val_accuracy: 0.3916\n"
+     ]
+    }
+   ],
+   "source": [
+    "# keras imports for the dataset and building our neural network\n",
+    "from keras.datasets import mnist\n",
+    "from keras.models import Sequential\n",
+    "from keras.layers import Dense, Dropout, Conv2D, MaxPool2D, Flatten\n",
+    "from keras.utils import np_utils\n",
+    "\n",
+    "# to calculate accuracy\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "\n",
+    "X = data\n",
+    "y = labels\n",
+    "\n",
+    "from sklearn.utils import shuffle\n",
+    "X, y = shuffle(X, y, random_state=0)\n",
+    "\n",
+    "\n",
+    "X_train = X[:int(len(X)*0.7)]\n",
+    "X_test = X[int(len(X)*0.7):]\n",
+    "\n",
+    "y_train = y[:int(len(X)*0.7)]\n",
+    "y_test = y[int(len(X)*0.7):]\n",
+    "\n",
+    "print(y_train[0])\n",
+    "print(X_train[0].shape)\n",
+    "\n",
+    "X_train = np.array(X_train).reshape(len(X_train), 313, 217, 3)\n",
+    "X_test = np.array(X_test).reshape(len(X_test), 313, 217, 3)\n",
+    "X_train = X_train.astype('float32')\n",
+    "X_test = X_test.astype('float32')\n",
+    "\n",
+    "y_train = np.array(y_train)\n",
+    "y_test = np.array(y_test)\n",
+    "\n",
+    "\n",
+    "# one-hot encoding using keras' numpy-related utilities\n",
+    "n_classes = 4\n",
+    "print(\"Shape before one-hot encoding: \", y_train.shape)\n",
+    "Y_train = np_utils.to_categorical(y_train, n_classes)\n",
+    "Y_test = np_utils.to_categorical(y_test, n_classes)\n",
+    "print(\"Shape after one-hot encoding: \", Y_train.shape)\n",
+    "\n",
+    "# building a linear stack of layers with the sequential model\n",
+    "model = Sequential()\n",
+    "# convolutional layer\n",
+    "model.add(Conv2D(25, kernel_size=(3,3), strides=(1,1), padding='valid', activation='relu', input_shape=(313,217,3)))\n",
+    "model.add(MaxPool2D(pool_size=(1,1)))\n",
+    "# flatten output of conv\n",
+    "model.add(Flatten())\n",
+    "# hidden layer\n",
+    "model.add(Dense(100, activation='relu'))\n",
+    "# output layer\n",
+    "model.add(Dense(4, activation='softmax'))\n",
+    "\n",
+    "# compiling the sequential model\n",
+    "model.compile(loss='categorical_crossentropy', metrics=['accuracy'], optimizer='adam')\n",
+    "\n",
+    "# training the model for 10 epochs\n",
+    "history = model.fit(X_train, Y_train, batch_size=128, epochs=20, validation_data=(X_test, Y_test))\n"
```

Comment:
Seems that here we only have train and test data. The test data is really being used as validation data. But really, we should have train, test, and validation data.

Reply:
Validation testing is not a necessary indicator; sufficiently allocated test data will suffice for the baseline. The split written by Rea was not used and can be deleted; it would make more sense to alter the model here if the need arises.